ci-operator/step-registry/ipi/conf/aws/custom-endpoints/ipi-conf-aws-custom-endpoints-commands.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+REGION="${LEASED_RESOURCE}"
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+function patch_endpoint()
+{
+  local service_name=$1
+  local service_endpoint=$2
+  local config_patch="${SHARED_DIR}/install-config-${service_name}.yaml.patch"
+  if [ "$service_endpoint" == "DEFAULT_ENDPOINT" ]; then
```

Comment:
How about when `service_endpoint` is "", we treat that as users want to default endpoint, once that, we can drop `DEFAULT_ENDPOINT` here.

Reply:
That means once users called `ipi-conf-aws-custom-endpoints`, but did not set a custom `SERVICE_ENDPOINT_*`, the script will set the endpoints to `https://${service_name}.${REGION}.amazonaws.com`

Reply:
`"DEFAULT_ENDPOINT"` means:
1. `serviceEndpoints.[serviceName]` field will be set in install-config.yaml
2. The endpoint value is  `https://${service_name}.${REGION}.amazonaws.com`

`""` means:
1. `serviceEndpoints.[serviceName]` field will NOT be set in install-config.yaml

We don't want to set EP in `install-config.yaml` if it is not added explicitly. 
For the following two configurations, some operators may have totally different behaviors (e.g. [OCPBUGS-1574](https://issues.redhat.com/browse/OCPBUGS-1574)), and we'd like to leave this control to the user.
* a) setting EP to the same as the default one in AWS SDK (`https://${service_name}.${REGION}.amazonaws.com`) explicitly in `install-config.yaml`
* b) don't set RP in config file.

Reply:
> "" means: serviceEndpoints.[serviceName] field will NOT be set in install-config.yaml

Sounds reasonable.